source/Patches/EndGameSummary.cs

```diff
@@ -0,0 +1,218 @@
+﻿using System;
+using System.Collections.Generic;
+//using Il2CppSystem.Collections.Generic;
+using System.Text;
+using UnityEngine;
+using Reactor;
+using TownOfUs.Roles;
+using TownOfUs.Roles.Modifiers;
+using System.Linq;
+using HarmonyLib;
+
+namespace TownOfUs.Patches
+{
+    [HarmonyPatch]
+    static class EndGameSummary
+    {
+        static class AdditionalTempData
+        {
+            public static List<PlayerInfo> playerData = new List<PlayerInfo>();
+
+            public static void clear()
+            {
+                playerData.Clear();
+            }
+
+            internal class PlayerInfo
+            {
+                public int PlayerId { get; set; }
+                public string PlayerName { get; set; }
+                public Faction Faction { get; set; }
+                public Role CurrentRole { get; set; }
+                public Modifier Modifier { get; set; }
+                public int TasksCompleted { get; set; }
+                public int TasksTotal { get; set; }
+            }
+        }
+
+        public static System.Collections.Generic.List<PlayerControl> allPlayers = new System.Collections.Generic.List<PlayerControl>();
+        public static bool playerInfoUpdateFlag1Updating = false;
+        public static bool playerInfoUpdateFlag2EndOfGame = false;
+
+        public static void UpdatePlayerDataRPCCall()
```

Comment:
Why is this needed?

Reply:
well this where i am baffled.

for just the end game summary, it could be called as the game ends, but that is not working, i placed it above the point where the winners setup, and it does not pull the data.

I was putting it in the tasks patch to be called whenever the task bar updates, and that works, but i think that is being called at the end of the game causing data to be only partial somehow? as i say, it was working fine in task patches until i changed something then it broke.

Reply:
i have fixed this, it now works within the task patch being run, this allows for constant updating meaning people can add additions (like task counts in meetings for dead people to see, etc etc) if it is desired, the problem was i put the assignment back to AdditionalTempData.PlayerData was being done within the loop, when it should have been after the loop, a misplace } on my part.